packages/pubsub/src/subscription.js

```diff
@@ -592,157 +442,398 @@ Subscription.prototype.decorateMessage_ = function(message) {
  *   var apiResponse = data[0];
  * });
  */
-Subscription.prototype.delete = function(callback) {
+Subscription.prototype.delete = function(gaxOpts, callback) {
   var self = this;
 
-  callback = callback || common.util.noop;
-
-  var protoOpts = {
-    service: 'Subscriber',
-    method: 'deleteSubscription'
-  };
+  if (is.fn(gaxOpts)) {
+    callback = gaxOpts;
+    gaxOpts = {};
+  }
 
   var reqOpts = {
     subscription: this.name
   };
 
-  this.parent.request(protoOpts, reqOpts, function(err, resp) {
-    if (err) {
-      callback(err, resp);
-      return;
+  this.request({
+    client: 'subscriberClient',
+    method: 'deleteSubscription',
+    reqOpts: reqOpts,
+    gaxOpts: gaxOpts
+  }, function(err, resp) {
+    if (!err) {
+      self.removeAllListeners('message');
+      self.close();
     }
 
-    self.closed = true;
-    self.removeAllListeners();
-
-    callback(null, resp);
+    callback(err, resp);
```

Comment:
> callback is supposed to be optional here. (I think?)

Ah, right!

> But also, if it wasn't successful, wouldn't we broadcast that to the error event listeners as we do in other places?

I would think no. The only places we emit errors are where it's likely they're coming from a Stream. In some places we have fall back code in case there is not an open stream, but that's only to provide consistent behavior with the most common use cases.

With regular rpcs like delete, I think it makes more sense to provide the error via callback.